assignments/array-methods.js

```diff
@@ -2,82 +2,509 @@
+// The event director needs both the first and last names of each runner for their running bibs.  Combine both the first and last names into a new array called fullName.
+const fullName = [];
+runners.forEach(runner => {
+  fullName.push(`${runner.first_name} ${runner.last_name}`);
+});
 console.log(fullName);
 
 // ==== Challenge 2: Use .map() ====
 // The event director needs to have all the runner's first names converted to uppercase because the director BECAME DRUNK WITH POWER. Convert each first name into all caps and log the result
-let allCaps = [];
-console.log(allCaps); 
+const allCaps = runners.map(runner => runner.first_name.toUpperCase());
+console.log(allCaps);
 
 // ==== Challenge 3: Use .filter() ====
 // The large shirts won't be available for the event due to an ordering issue.  Get a list of runners with large sized shirts so they can choose a different size. Return an array named largeShirts that contains information about the runners that have a shirt size of L and log the result
-let largeShirts = [];
+const largeShirts = runners.filter(runner => {
+  return runner.shirt_size === "L";
+});
 console.log(largeShirts);
 
 // ==== Challenge 4: Use .reduce() ====
 // The donations need to be tallied up and reported for tax purposes. Add up all the donations into a ticketPriceTotal array and log the result
-let ticketPriceTotal = [];
+const ticketPriceTotal = runners.reduce((total, runner) => {
+  return total + runner.donation;
+}, 0);
 console.log(ticketPriceTotal);
 
 // ==== Challenge 5: Be Creative ====
 // Now that you have used .forEach(), .map(), .filter(), and .reduce().  I want you to think of potential problems you could solve given the data set and the 5k fun run theme.  Try to create and then solve 3 unique problems using one or many of the array methods listed above.
 
 // Problem 1
 
+// The community center wants send a special thank you letter to the runners who represent a business that donated $100 or more. Create an array that includes only such runners.
+const bigSpenders = runners.filter(runner => {
+  return runner.donation > 99;
+});
+console.log(bigSpenders);
+
 // Problem 2
 
-// Problem 3
\ No newline at end of file
+// A group of hackers wants to send spam emails to all the runners. These strange hackers don't care about the runners' names or anything else - all they want is an array of email addresses. Create the array.
+const justEmailAddresses = runners.map(runner => runner.email);
+console.log(justEmailAddresses);
+
+// Problem 3
+// It turns out that all shirts size large and above are unavailable - what a disaster! Get a list of runners who chose size large or above. Create an array named shirtContact that contains ONLY first_name, last_name, email, and shirt_size so those runners can be contacted later when their shirts arrive.
+const shirtContact = runners.filter(runner => {
+  delete runner.id;
+  delete runner.company_name;
+  delete runner.donation;
+  return runner.shirt_size.includes("L");
```

Comment:
clever way of doing this with using `delete` to get rid of the data you don't want instead of just carrying over the data you do